wallet-new/docs/spec.tex

```diff
@@ -3009,38 +3025,533 @@ \subsection{Use cases}
+for a 10:0 deposit:payment ratio.
+
+\subsubsection{Real data}
+
+\paragraph{\texttt{MoneyPot.com}}
+
+Ideally, of course, we run the simulation against \emph{real} event streams from
+existing wallets. Unfortunately, such data is hard to come by. Erhardt was able
+to find one such dataset, provided by \texttt{MoneyPot.com}
+\citep{MoneyPotdataset}. Figure~\ref{fig:inputselection:MoneyPot_randomOn} shows
+the results of running our algorithm on this dataset.
+
+\begin{figure}[p]
+\begin{center}
+\scriptsize
+\begin{tabular}{ll}
+\input{spec/MoneyPot-randomOn.tex} &
+\input{spec/MoneyPot-randomOn-distr.tex}
+\end{tabular}
+\end{center}
+\caption{\label{fig:inputselection:MoneyPot_randomOn}
+  \texttt{Random-Improve}, using the MoneyPot data set. There is a rougly 2:1
+  deposit:payment ratio. Values have been scaled. Log scale on the x-axis.
+}
+\end{figure}
+
+A few observations are in order here. First, there are quite a few deposits and
+payments close to 0, just like in an exponential distribution. Moreover,
+although we have many values close to 0, we also have some \emph{huge} outliers; the
+payments are closely clustered together, but there is a $10^9$ difference between
+the smallest and largest deposit, and moreover a $10^5$ difference between the
+largest deposit and the largest payment. It is therefore not surprising that we
+end up with a relatively large change:payment ratio. Nonetheless, the algorithm
+is behaving well, with the size of the UTxO tracking the balance nicely, with an
+average UTxO size of 130 entries. The average number of outputs is 3.03, with
+50\% of transactions using just one input, and 90\% using 6 or fewer.
+
+\paragraph{Cardano Exchange}
+
+One of the large Cardano exchange node has also helped us with
+some anonymised data (deposits and payments), similar in nature to the
+MoneyPot dataset albeit significantly larger. Coming from an exchange node,
+however, this dataset is \emph{very much} skewed towards deposits, with a
+deposit:payment ratio of roughly 30:1. Under these circumstances, of course,
+coin selection alone cannot keep the UTxO size small
+(Figure~\ref{fig:inputselection:CardanoExchange_randomOn}).
+
+\begin{figure}[p]
+\begin{center}
+\scriptsize
+\begin{tabular}{ll}
+\input{spec/CardanoExchange-randomOn.tex} &
+\input{spec/CardanoExchange-randomOn-distr.tex}
+\end{tabular}
+\end{center}
+\caption{\label{fig:inputselection:CardanoExchange_randomOn}
+  \texttt{Random-Improve}, using data set from a large Cardano exchange.
+  There is a rougly 30:1 deposit:payment ratio. Values have been scaled.
+  Log scale on the x-axis.
+}
+\end{figure}
+
+The UTxO here also contains some truly enormous outputs, which is pushing the
+change:payment ratio up, although as more withdrawals are being made the
+algorithm is able to push it down. Coin selection alone cannot deal with
+a situation like this (and indeed, exchanges have additional infrastructure in
+place that alleviates this problem).
```

Comment:
Perhaps here's the place to mention what additional trick we would recommend in this case, ie maintenance txs: given the distributions what should those maintance txs look like, and why would we expect it to have the intended effect.

The fact that exchanges already do something is mildly interesting, but with our scheme here we can say what we recommend that would be complementary.